packages/webpack-config-single-spa/lib/webpack-config-single-spa.js

```diff
@@ -28,6 +28,10 @@ function webpackConfigSingleSpa(opts) {
 
   let webpackConfigEnv = opts.webpackConfigEnv || {};
 
+  let argv = opts.argv || {};
```

Comment:
The Webpack docs seem to lean into using the `env` (or rather `webpackConfigEnv` in our case) parameter to read the production flag rather than directly off of `argv`, as seen here: https://webpack.js.org/configuration/configuration-types/#exporting-a-function. I don't know the difference well enough. Is there a reason to prefer `argv` over `env`?

Reply:
The webpack env doesn't specify whether it's production mode or not unless you add `--env.prod` to the webpack command. This is a bit redundant, since we already are doing `webpack --mode=production` in the package.json scripts. The `argv` are the arguments passed to webpack's CLI, including the `--mode=production`. Webpack CLI parses the argv with `yargv` in webpack 4 and `commander` in webpack 5.

The reason I chose argv here is to avoid the redundancy of having `webpack --mode=production --env.prod`

Reply:
Ah ok, thanks for explaining! I didn't notice they're different flags.

Reply:
Yeah the `--mode` flag is one supported by webpack itself, whereas anything starting with `--env` is ignored by webpack but passed onto the webpack configs. In this case, we want to use the `--mode` flag so we get the default webpack behavior for that, but also want the webpack configs to be aware of it.